packages/grafana-ui/src/components/Forms/Form.mdx

```diff
@@ -62,6 +62,67 @@ Register accepts an object which describes validation rules for a given input:
 />
 ```
 
+##### Custom validation rules
+
+When registering a form element it is possible to supply custom validation rules. `validate` property of `register` (as well as of `rules`) accepts
+a callback function or an object of callbacks that apply custom validation rules.
+
+The type signature of `validate` callback is:
+```jsx
+type ValidateResult = string | boolean | undefined;
+type Validate = (data: any) => ValidateResult | Promise<ValidateResult>;
+```
+
+According to this signature, validation callback can also return a string, used as an error message.
+Some examples of custom validation (taken from react-hook-form [documentation](https://react-hook-form.com/api/#register)):
```

Comment:
This is more like blog content rather than docs. I would skip it and keep this more informative rather than explanatory.